src/rubicon/objc/api.py

```diff
@@ -2182,7 +2195,35 @@ def __call__(self, *args):
         objects according to the default ``ctypes`` rules, based on the
         block's return and parameter types.
         """
-        return self.struct.contents.invoke(self.pointer, *args)
+        # If any of the arguments are structures, they may be anonymous - that is, we
+        # have a descriptor like "{=ii}", which tells us there are two integer fields,
+        # but doesn't provide a name for the structure. ctypes looks for an exact match
+        # of type names, so even if the field types of a structure provided as an
+        # argument match, ctypes will raise a TypeError.
+        #
+        # To avoid this, look for the `__anonymous__` property on the structure
+        # definition. If it exists, check that the provided argument matches the "shape"
+        # of the anonymous structure. If it matches, modify the invoke signature to
+        # match the type of the argument that was actually provided. The first argument
+        # to invoke is the block being invoked, so we can ignore that type hint.
+        for i, argtype in enumerate(self.invoke_argtypes[1:]):
+            if hasattr(argtype, "__anonymous__"):
```

Comment:
```suggestion
            if getattr(argtype, "__anonymous__", False):
```

Reply:
If I understand correctly, there is no need manually check the field types if this is a named structure, right?

Reply:
Also, maybe add a check that we are indeed dealing with a structure? Using the presence of an `__anonymous__` field for this is very implicit requires knowledge of when this field is set.

Reply:
You're correct that no type check is required if it is a named structure. If the block explicitly declares its argument as an NSPoint, ctypes will accept an NSPoint. It's only when the block anonymously declares that it accepts "a structure that contains 2 double fields" that we need to check the shape and modify the argtype so that ctypes won't generate a TypeError.

Explicitly checking for a Structure type is definitely worthwhile - ``__anonymous__`` *should* only exist on structures, but it doesn't hurt to be more explicit.